src/heitech.zer0mqXt.core/Patterns/RqRep.cs

```diff
@@ -18,33 +17,12 @@ internal RqRep(SocketConfiguration configuration)
         }
 
         #region Request / Client
-        public async Task<XtResult<TResult>> RequestAsync<T, TResult>(T request)
+        public Task<XtResult<TResult>> RequestAsync<T, TResult>(T request)
             where T : class, new()
             where TResult : class, new()
         {
-            try
-            {
-                return await DoRequestAsync<T, TResult>(request).ConfigureAwait(false);
-            }
-            catch (NetMQ.EndpointNotFoundException ntfnd)
-            {
-                _configuration.Logger.Log(new ErrorLogMsg($"NetMQ.Endpoint could not be found at {_configuration.Address()}: " + ntfnd.Message));
-                await Task.Delay((int)_configuration.TimeOut.TotalMilliseconds).ConfigureAwait(false);
-                try
-                {
-                    return await DoRequestAsync<T, TResult>(request).ConfigureAwait(false);
-                }
-                catch (System.Exception inner)
-                {
-                    _configuration.Logger.Log(new ErrorLogMsg("Request failed after Retry: " + inner.Message));
-                    return XtResult<TResult>.Failed(inner);
-                }
-            }
-            catch (System.Exception ex)
-            {
-                _configuration.Logger.Log(new ErrorLogMsg("Request failed: " + ex.Message));
-                return XtResult<TResult>.Failed(ex);
-            }
+            var retry = new Retry(_configuration);
+            return retry.RunAsyncWithRetry<TResult>(async () => await DoRequestAsync<T, TResult>(request));
```

Comment:
_async / await_ isn't really required here.
But if we'd still like to use it, we may want to add `.ConfigureAwait(false)` after the invocation of `DoRequestAsync` here as well.

Reply:
done (removed async & await)